pkg/build/api/types.go

```diff
@@ -76,19 +87,71 @@ type BuildSpec struct {
 	// Output describes the Docker image the Strategy should produce.
 	Output BuildOutput
 
-	// Compute resource requirements to execute the build
+	// Resources computes resource requirements to execute the build.
 	Resources kapi.ResourceRequirements
 
 	// PostCommit is a build hook executed after the build output image is
 	// committed, before it is pushed to a registry.
 	PostCommit BuildPostCommitSpec
 
-	// Optional duration in seconds, counted from the time when a build pod gets
-	// scheduled in the system, that the build may be active on a node before the
-	// system actively tries to terminate the build; value must be positive integer
+	// CompletionDeadlineSeconds is an optional duration in seconds, counted from
+	// the time when a build pod gets scheduled in the system, that the build may
+	// be active on a node before the system actively tries to terminate the
+	// build; value must be positive integer.
 	CompletionDeadlineSeconds *int64
 }
 
+// BuildTriggerCause holds information about the trigger of a build.
+type BuildTriggerCause struct {
+	// Reason is used to store a human readable reason for why the build was
+	// triggered. E.g.: "Build manually triggered by user", "Build triggered by
+	// ConfigChange update", "Build triggered by ImageChange for image:
+	// FOO:BAR", etc.
+	Reason string
```

Comment:
@smarterclayton i can see your reasoning, i'll rename it to Message. It's intended to be a field that holds a printable sentence that says what triggered the build. "image change", "build config", "config change", etc. Then i can say ``` Build trigger cause:    GitHub webhook ```